packages/quantic/cypress/integration/case-classification/case-classification.cypress.ts

```diff
@@ -0,0 +1,424 @@
+    it('should display the loading spinner', () => {
+      visitCaseClassification({
+        maxChoices: defaultMaxChoices,
+      });
+
+      scope('when loading the page', () => {
+        Expect.displayLabel(true);
+        Expect.numberOfInlineOptions(0);
+        Expect.numberOfSuggestions(0);
+        Expect.displaySelectTitle(false);
+        Expect.displaySelectInput(true);
+      });
+
+      scope('when fetching suggestions', () => {
+        const suggestionsCount = 2;
+        mockCaseClassification(
+          coveoDefaultField,
+          allOptions.slice(0, suggestionsCount)
+        );
+        interceptClassificationsIndefinitely();
+        fetchClassifications();
+        Expect.displaySelectTitle(false);
+        Expect.displaySelectInput(false);
+        Expect.numberOfSuggestions(0);
+        Expect.numberOfInlineOptions(0);
+        Expect.displayLoading(true);
+      });
+    });
+  });
+
+  describe('when no suggestions are found', () => {
+    it('should display only the select dropdown', () => {
+      visitCaseClassification({
+        maxChoices: defaultMaxChoices,
+      });
+
+      scope('when loading the page', () => {
+        Expect.numberOfInlineOptions(0);
+        Expect.numberOfSuggestions(0);
+        Expect.displaySelectTitle(false);
+        Expect.displaySelectInput(true);
+      });
+
+      scope('when fetching suggestions', () => {
+        mockCaseClassification(coveoDefaultField, []);
+        fetchClassifications();
+        Expect.displaySelectTitle(false);
+        Expect.displaySelectInput(true);
+        Expect.numberOfSuggestions(0);
+        Expect.numberOfInlineOptions(0);
+      });
+    });
+  });
+
+  describe('when maxChoices set to 1', () => {
+    it('should display only the select dropdown', () => {
+      visitCaseClassification({
+        maxChoices: 1,
+      });
+
+      scope('when loading the page', () => {
+        Expect.displayLabel(true);
+        Expect.numberOfInlineOptions(0);
+        Expect.numberOfSuggestions(0);
+        Expect.displaySelectTitle(false);
+        Expect.displaySelectInput(true);
+      });
+
+      scope('when fetching suggestions', () => {
+        const suggestionsCount = 2;
+        mockCaseClassification(
+          coveoDefaultField,
+          allOptions.slice(0, suggestionsCount)
+        );
+        fetchClassifications();
+        Expect.displaySelectTitle(false);
+        Expect.numberOfSuggestions(0);
+        Expect.numberOfInlineOptions(0);
+        Expect.displaySelectInput(true);
+      });
+
+      scope('when selecting an option from the select input', () => {
+        const clickedIndex = 3;
+
+        Actions.openSelectInput();
+        Actions.clickSelectOption(clickedIndex);
+        Expect.logUpdatedClassificationFromSelectOption(
+          coveoDefaultField,
+          clickedIndex
+        );
+      });
+    });
+  });
+
+  describe('when invalid number of maxChoices is given', () => {
```

Comment:
would love to have a test to cover the situation when user set `invalid Salesforce field`